### PR TITLE
dolphin-emu: 5.0-19870 -> 5.0-20347

### PR DESCRIPTION
**Pull request by qubitnano**
## Description of changes

https://dolphin-emu.org/blog/2023/11/25/dolphin-progress-report-august-september-and-october-2023/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by majiru** [comment]
Result of `nixpkgs-review pr 279439` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dolphin-emu</li>
  </ul>
</details>

**Review by majiru — APPROVED** [review]

**Comment by majiru** [comment]
When testing with this locally, I noticed that the `find-minizip-ng.patch` seems to also be no longer required. At least I got everything to build and run without it.

**Comment by KamilaBorowska** [comment]
> When testing with this locally, I noticed that the `find-minizip-ng.patch` seems to also be no longer required. At least I got everything to build and run without it.

I didn't check it yet, but I believe that without this patch Dolphin uses a vendored version of zlib-ng.

**Comment by majiru** [comment]
> > When testing with this locally, I noticed that the `find-minizip-ng.patch` seems to also be no longer required. At least I got everything to build and run without it.
> 
> I didn't check it yet, but I believe that without this patch Dolphin uses a vendored version of zlib-ng unless something did change in this release.

Ah okay thanks. Is there an easy way we could test to see what winds up vendored and what winds up not?

File: pkgs/applications/emulators/dolphin-emu/default.nix [hunk 1]
```diff
@@ -57,21 +57,17 @@
...
   pname = "dolphin-emu";
-  version = "5.0-19870";
+  version = "5.0-20347";
 
   src = fetchFromGitHub {
     owner = "dolphin-emu";
     repo = "dolphin";
-    rev = "032c77b462a220016f23c5079e71bb23e0ad2adf";
-    sha256 = "sha256-TgRattksYsMGcbfu4T5mCFO9BkkHRX0NswFxGwZWjEw=";
+    rev = "dc0814ae4622313d513468bdc377ee9c031de199";
```
Comment (a-n-n-a-l-e-e) [inline pkgs/applications/emulators/dolphin-emu/default.nix:65]: how do you match a version to a rev?

**Comment by a-n-n-a-l-e-e** [comment]
putting a link to the changes / changelog is common and helpful to reviewers.

File: pkgs/applications/emulators/dolphin-emu/default.nix (hunk 1 above)
Comment (qubitnano) [reply pkgs/applications/emulators/dolphin-emu/default.nix:65]: `nix-shell maintainers/scripts/update.nix --argstr package dolphin-emu` or check https://dolphin-emu.org/download/

**Review by ghost — APPROVED** [review]
builds, runs, LGTM

Result of `nixpkgs-review pr 279439` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dolphin-emu</li>
  </ul>
</details>

**Pull request merged**